composer.json

```diff
@@ -23,12 +23,11 @@
     "require": {
         "php": "^7.2",
         "doctrine/dbal": "^2.6",
-        "doctrine/persistence": "^1.3.6"
+        "doctrine/persistence": "^1.3.6 || ^2.0"
     },
     "require-dev": {
-        "doctrine/common": "^2.7",
+        "doctrine/common": "^2.7 || ^3.0",
         "doctrine/orm": "^2.5",
-        "doctrine/phpcr-odm": "^1.4",
```

Comment:
Why do we need to make that dev requirement dynamic?

Reply:
Because "doctrine/phpcr-odm:1.x" requires "doctrine/common:2.x", and "doctrine/common:2.x" requires "doctrine/persistence:1.x":
```
Problem 1
    - doctrine/common v2.9.0 requires doctrine/persistence ^1.0 -> satisfiable by doctrine/persistence[1.0.x-dev, 1.1.1, 1.1.x-dev, 1.2.0, 1.2.x-dev, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev, v1.0.0, v1.0.1, v1.1.0] but these conflict with your requirements or minimum-stability.
    - doctrine/common v2.10.0 requires doctrine/persistence ^1.1 -> satisfiable by doctrine/persistence[1.1.1, 1.1.x-dev, 1.2.0, 1.2.x-dev, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev, v1.1.0] but these conflict with your requirements or minimum-stability.
    - doctrine/common v2.11.0 requires doctrine/persistence ^1.1 -> satisfiable by doctrine/persistence[1.1.1, 1.1.x-dev, 1.2.0, 1.2.x-dev, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev, v1.1.0] but these conflict with your requirements or minimum-stability.
    - doctrine/common 2.13.3 requires doctrine/persistence ^1.3.3 -> satisfiable by doctrine/persistence[1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev] but these conflict with your requirements or minimum-stability.
    - doctrine/common 2.13.2 requires doctrine/persistence ^1.3.3 -> satisfiable by doctrine/persistence[1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev] but these conflict with your requirements or minimum-stability.
    - doctrine/common 2.13.1 requires doctrine/persistence ^1.3.3 -> satisfiable by doctrine/persistence[1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev] but these conflict with your requirements or minimum-stability.
    - doctrine/common 2.13.0 requires doctrine/persistence ^1.3.3 -> satisfiable by doctrine/persistence[1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev] but these conflict with your requirements or minimum-stability.
    - doctrine/common 2.12.0 requires doctrine/persistence ^1.1 -> satisfiable by doctrine/persistence[1.1.1, 1.1.x-dev, 1.2.0, 1.2.x-dev, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.3.5, 1.3.6, 1.3.7, 1.3.x-dev, 1.4.x-dev, v1.1.0] but these conflict with your requirements or minimum-stability.
    - Conclusion: remove doctrine/persistence 2.0.0
    - doctrine/common v2.7.1 conflicts with doctrine/persistence[2.0.0].
    - doctrine/common v2.7.2 conflicts with doctrine/persistence[2.0.0].
    - doctrine/common v2.7.3 conflicts with doctrine/persistence[2.0.0].
    - doctrine/common v2.8.0 conflicts with doctrine/persistence[2.0.0].
    - doctrine/common v2.8.1 conflicts with doctrine/persistence[2.0.0].
    - doctrine/common v2.7.0 conflicts with doctrine/persistence[2.0.0].
    - Installation request for doctrine/persistence ^2.0 -> satisfiable by doctrine/persistence[2.0.0].
    - Installation request for doctrine/common ^2.7 -> satisfiable by doctrine/common[2.12.0, 2.13.0, 2.13.1, 2.13.2, 2.13.3, v2.10.0, v2.11.0, v2.7.0, v2.7.1, v2.7.2, v2.7.3, v2.8.0, v2.8.1, v2.9.0].
```

Reply:
Maybe wait for this then: https://github.com/doctrine/mongodb-odm/pull/2204 ?

Reply:
I've added a new commit with the suggested approach.

Reply:
Oh sorry I've been confusing mongodb-odm and phpcr-odm… I don't think it makes sense to use one instead of of the other given that there are PHPCR namespaces in our tests.

Reply:
IMO we should make phpcr-odm compatible instead of removing the dependency

Reply:
>IMO we should make phpcr-odm compatible instead of removing the dependency

That would be great, but since the model implementation is not enforced to be specific; I'd like to focus this PR in make this package compatible with "doctrine/persistence:^2.x". Users using "doctrine/phpcr-odm" will be still able to use "doctrine/persistence:^1.x", while the users using other implementation would be able to leverage "doctrine/persistence:^2.x".
When "doctrine/phpcr-odm" reaches its compatibility with "doctrine/persistence:^2.x", we should revert the condition added here.

To be clear, if we don't make the dependency conditional in our tests, we aren't able to install "doctrine/persistence:^2.x" for other model managers aren't part of this conflict.

Reply:
FYI I just created https://github.com/doctrine/phpcr-odm/pull/812to address deprecations, if it gets merged I will do another to add support for persistence 2 (which implies common 3)

Reply:
phpcr-odm 1.5.2 now has support for persistence 2 and common 3 :)

Reply:
Thank you @greg0ire